src/main/java/de/berlin/lostberlin/service/BusinessServiceImpl.java

```diff
@@ -65,15 +66,40 @@ public Business saveUpdatedBusinessProfile(Long id, BusinessUpdateDto businessPr
         if (business.getEMail().equals(businessProfile.getEMail()) && businessRepo.existsByEMail(businessProfile.getEMail())) {
             throw new EntityNotUniqueException("Email already exists");
         }
-        business.setFName(businessProfile.getFName());
-        business.setLName(businessProfile.getLName());
-        business.setEMail(businessProfile.getEMail());
-        business.setPhone(businessProfile.getPhone());
-        business.setDescription(businessProfile.getDescription());
-        business.setServiceLocation(businessProfile.getServiceLocation());
-        business.setPhoto(businessProfile.getPhoto());
-        business.setUsername(businessProfile.getUsername());
+        if(businessProfile.getFName()!=null){
+        business.setFName(businessProfile.getFName());}
+
+        if(businessProfile.getLName()!=null){
+        business.setLName(businessProfile.getLName());}
+
+        if(businessProfile.getEMail()!=null){
+        business.setEMail(businessProfile.getEMail());}
 
+        if(businessProfile.getPhone()!=null){
+        business.setPhone(businessProfile.getPhone());}
```

Comment:
All these checks for nulls are nasty. Why not just setting the value if the method is called? What will go wrong in case you do not check for null?

Reply:
I added them to prevent overwriting existing values with nulls (for example, if the user only updates a phone number, everything else is overwritten with null). I checked for alternatives and found json patch, should I implement it or are there better ways to handle this?

Reply:
PATCH is more suitable here since it is about update a part of the resource and it is what we need.

Reply:
If you cannot avoid null checks (need to think about a better solution though), move them to a separate method with ?xx:you syntax

Reply:
I use patch for updating file path, since it will happen each time a profile is created. But I can also think of a case when a user wants to updated his whole profile or just parts of it. I didn't want to create a bunch of patch methods and DTOs for this.

Reply:
Sorry, what's a ?xx:you syntax? What do you think about Json Patch solution: https://williamdurand.fr/2014/02/14/please-do-not-patch-like-an-idiot/

Reply:
I meant instead of if(){} use a?b:c syntax and move this to a separate private method. 

Using merge patch looks interesting, but you will have to implement its processing on the server side as well as generation of such json on the client side which I am not sure is easy and there are libraries. So, you may end up with a bit of overhead to support it. 

May be using put is also fine assuming that null values are non-preset values and should not replace its counterparts in the original resource.

Reply:
actually,I had similar concerns. So I'm relieved:)

I can move the if-checks to a separate method, but why using ternary operator when I have only one choice? Threre's no "then" in those checks.

Reply:
You’re right. Let’s move it out first. I still think we can avoid such explicit checks. Will see later.

Reply:
ok, thanks!